packages/keystores-browser/src/multi_contract_browser_local_storage_key_store.ts

```diff
@@ -0,0 +1,159 @@
+import { KeyPair, KeyPairString } from '@near-js/crypto';
+import { MultiContractKeyStore } from '@near-js/keystores';
+
+const LOCAL_STORAGE_KEY_PREFIX = 'near-api-js:keystore:';
+
+/**
+ * This class is used to store keys in the browsers local storage.
+ * 
+ * @see [https://docs.near.org/docs/develop/front-end/naj-quick-reference#key-store](https://docs.near.org/docs/develop/front-end/naj-quick-reference#key-store)
+ * @example
+ * ```js
+ * import { connect, keyStores } from 'near-api-js';
+ * 
+ * const keyStore = new keyStores.MultiContractBrowserLocalStorageKeyStore();
+ * const config = { 
+ *   keyStore, // instance of MultiContractBrowserLocalStorageKeyStore
+ *   networkId: 'testnet',
+ *   nodeUrl: 'https://rpc.testnet.near.org',
+ *   walletUrl: 'https://wallet.testnet.near.org',
+ *   helperUrl: 'https://helper.testnet.near.org',
+ *   explorerUrl: 'https://explorer.testnet.near.org'
+ * };
+ * 
+ * // inside an async function
+ * const near = await connect(config)
+ * ```
+ */
+export class MultiContractBrowserLocalStorageKeyStore extends MultiContractKeyStore {
+    /** @hidden */
+    private localStorage: Storage;
+    /** @hidden */
+    private prefix: string;
+
+    /**
+     * @param localStorage defaults to window.localStorage
+     * @param prefix defaults to `near-api-js:keystore:`
+     */
+    constructor(localStorage: any = window.localStorage, prefix = LOCAL_STORAGE_KEY_PREFIX) {
+        super();
+        this.localStorage = localStorage;
+        this.prefix = prefix;
```

Comment:
Can we please make this `this.prefix = prefix || LOCAL_STORAGE_PREFIX;`?

In the current implementation, an empty string prefix means `clear()` would delete all localstorage entries.

Reply:
Let's merge the original PR into a feature branch.

Reply:
@gtsonevv go ahead and rebase your changes on top of #1370 and we can merge that once your changes are in.